### DISCUSSION TITLE
Is it possible to debug multiple-theaded applications?

### DISCUSSION THREAD

**Discussion by TheLeoP** [mfussenegger/nvim-jdtls#266] [Q&A]
I'm really lost about this. I think [this](https://github.com/microsoft/java-debug/pull/315) provides support for multiple threaded debuging, but I'm not really sure if dap-nvim even supports this (?). Is this possible? Am I missing something? Thanks, in advance.

EDIT: the link references multiple threaded debbugin capabilities for the java-debug extension (I think?)

**Comment by mfussenegger** [comment]
nvim-jdtls currently doesn't support any of the linked out-of-spec requests, but nvim-dap in general supports debugging of multiple threads - with the limitation that only a single thread can be stopped at the breakpoint at a time (others can be paused).

See https://github.com/mfussenegger/nvim-dap/pull/498 for a demo
